test/libsolidity/smtCheckerTests/bmc_coverage/range_check.sol

```diff
@@ -0,0 +1,57 @@
+pragma experimental SMTChecker;
+
+contract C {
+	constructor() payable {
+		assert(tx.origin >= address(0));
+		assert(tx.origin <= address(2**160 - 1));
+		assert(tx.gasprice >= 0);
+		assert(tx.gasprice <= 2**256 - 1);
+		assert(msg.sender >= address(0));
+		assert(msg.sender <= address(2**160 - 1));
+		assert(msg.value >= 0);
+		assert(msg.value <= 2**256 - 1);
+
+		assert(block.coinbase >= address(0));
+		assert(block.coinbase <= address(2**160 - 1));
+		assert(block.timestamp >= 0);
+		assert(block.timestamp <= 2**256 - 1);
+		assert(block.chainid >= 0);
+		assert(block.chainid <= 2**256 - 1);
+		assert(block.difficulty >= 0);
+		assert(block.difficulty <= 2**256 - 1);
+		assert(block.gaslimit >= 0);
+		assert(block.gaslimit <= 2**256 - 1);
+		assert(block.number >= 0);
+		assert(block.number <= 2**256 - 1);
+	}
+}
+
+contract D {
+	constructor() payable {
+		unchecked {
+			assert(tx.origin >= address(0));
```

Comment:
So `unchecked` blocks are essentially disabling SMT checker?

Reply:
It is not disabled, but there is a difference in encoding, see, e.g., https://github.com/ethereum/solidity/pull/10704#issuecomment-769304388

Basically, in checked blocks type constraints are automatically added for intermediate expressions, which as a side effect constrains also transaction variables. This change has been made after the first version of this PR.